### PR TITLE
Implement kmath-nd4j: module that implements NDStructure for INDArray of ND4J

### LINKED ISSUE #52: ND4J
Add implementation of `NDStructure` with ND4J

### PR DESCRIPTION
**Pull request by CommanderTvis**
closes #52

### CODE REVIEW

**Comment by altavir** [comment]
Please resolve conflicts

**Comment by CommanderTvis** [comment]
@altavir

File: kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+import kscience.kmath.operations.*
+import kscience.kmath.structures.*
+
+/**
+ * Represents [NDAlgebra] over [INDArrayAlgebra].
+ *
+ * @param T the type of ND-structure element.
+ * @param C the type of the element context.
+ */
+public interface INDArrayAlgebra<T, C> : NDAlgebra<T, C, INDArrayStructure<T>> {
```
Comment (altavir) [inline kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt:14]: I do not like naming. Could we use `Nd4jArrayAlgebra` etc?

**Review by altavir — APPROVED** [review]
Looks nice. Is it possible to provide a test where ND4J structures are used alongside regular Real/Buffered NDAlgebra elements?

File: kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt (hunk 1 above)
Comment (CommanderTvis) [reply kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt:14]: Yes

Comment (CommanderTvis) [reply kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt:14]: It is 5 minutes to refactor.

**Review by altavir — APPROVED** [review]

**Pull request merged**